docs/proposals/14-wireguard-based-cluster-vpn.md

```diff
@@ -0,0 +1,178 @@
+
+# Cluster VPN based on Wireguard with reversed connection
+
+# Problem Statement
+
+Today's Gardener cluster VPN solution has several major issues including:
+
+1. Connection establishment is always from the seed cluster to the shoot cluster. This means that there needs to be connectivity both ways which is not desirable in many cases (OpenStack, VMware) and causes high effort in firewall configuration or extra infrastructure. These firewall configurations are prohibited in some cases due to security policies.
+2. Shoot clusters must provide a VPN endpoint. This means extra cost for the endpoint (roughly €20/month on hyperscalers) or will consume scarce resources (limited number of VMware NSX-T load balancers).
+
+Issue 1 above currently prevents us from rolling out our infrastructure efficiently on OpenStack.
```

Comment:
The main issue is that the landscapes might be rather fenced. Each domain has only limited or no connectivity to other domains. It might be possible to have one domain being accessible from multiple other domains, but it is not possible for one domain to connect to all other.
Currently, we need connectivity from shoot to seed (ordinary operation) and seed to shoot (vpn shoot).
This means that we could operate one seed cluster per domain, but this is hardly efficient as we expect the seeds in this case to only be lightly loaded. Furthermore, this would mean that we might need to operate a lot of seeds as we see potential scale in terms of domains.
In case we can ensure that we only require connectivity in the direction shoot to seed, we could operate only a single seed (or overall less seeds) for multiple domains.
@dkistner might be able to provide more details.

Reply:
Yes, exactly as @ScheererJ has described it.
Image an infrastructure where you want to have a central seed for shoots which are running in different network environments, that can't be accessed by the Seed (shoot can call to seed, seed can't call to shoot). In such a scenario you would need with the current vpn solution a Seed in each of those network environments, which is not very "effective" from resource consumption/maintenance point of view.

Reply:
Thanks! This is more or less is valid general statement for improving the experience Gardener landscapes where the seed and shoots can be in completely different fenced environments with only prerequisite the shoot to have network connectivity the public network of the seed.

I.e. it is nothing OpenStack specific and it will be nice the above statement to be rephrased.

Reply:
Yes, it is not strictly open stack specific, but rather specific to fenced environments. The statement can be improved.

Reply:
I will remove that sentence altogether. The description on 1. is sufficient to justify the need without exposing enterprise specific setups.